restli-server/src/test/java/com/linkedin/restli/internal/server/TestRestLiRouter.java

```diff
@@ -0,0 +1,237 @@
+              clazz.getSimpleName(), t.getMessage()), t);
+    }
+    throw new IllegalStateException(
+        String.format("Expected an exception of type '%s' but none was caught", clazz.getSimpleName()));
+  }
+
+  @RestLiCollection(name = "root", keyName = "rootId")
+  private static class RootResource extends CollectionResourceTemplate<Long, EmptyRecord>
+  {
+    @RestMethod.Get
+    public EmptyRecord get(@PathKeyParam("rootId") Long id)
+    {
+      return new EmptyRecord();
+    }
+  }
+
+  private static final class TestSetup
+  {
+    private final String _childPath;
+    private final RestLiConfig _config;
+    private final ServerResourceContext _context;
+    private final Class<?> _keyClass;
+    private final String _keyName;
+    private final DataMap _parameters;
+    private final MutablePathKeys _pathKeys;
+    private final Map<String, ResourceModel> _pathToModelMap;
+    private final String _resourceName;
+    private final RequestContext _requestContext;
+    private final ResourceModel _rootModel;
+    private final String _rootPath;
+    private final RestLiRouter _router;
+
+    private TestSetup() {
+      _keyName = "rootId";
+      _keyClass = Long.class;
+      _resourceName = "root";
+      _childPath = "/child";
+      _rootPath = "/root";
+      _parameters = new DataMap();
+      _context = mock(ServerResourceContext.class);
+      _pathKeys = mock(MutablePathKeys.class);
+      _rootModel = RestLiAnnotationReader.processResource(RootResource.class);
+      _pathToModelMap = new HashMap<>();
+      _config = new RestLiConfig();
+      _router = new RestLiRouter(_pathToModelMap, _config);
+      _requestContext = new RequestContext();
+    }
+
+    private void mockCommon() {
+      doReturn(_parameters).when(_context).getParameters();
+      doReturn(_pathKeys).when(_context).getPathKeys();
+      doReturn(RESTLI_PROTOCOL_2_0_0.getProtocolVersion()).when(_context).getRestliProtocolVersion();
+      doReturn(_requestContext).when(_context).getRawRequestContext();
+      _pathToModelMap.put(_rootPath, _rootModel);
+    }
+
+    private void mockContextForRootResourceGetRequest(String path) throws URISyntaxException
+    {
+      mockCommon();
+      doReturn(null).when(_pathKeys).getBatchIds();
+      doReturn(new URI(path)).when(_context).getRequestURI();
+      doReturn("GET").when(_context).getRequestMethod(); // http method
+      doReturn("GET").when(_context).getRestLiRequestMethod(); // header value
+      doReturn(null).when(_context).getMethodName(eq(ResourceMethod.GET));
+    }
+
+    private void mockContextForOperationNotFound(String path) throws URISyntaxException
+    {
+      mockCommon();
+      doReturn(null).when(_pathKeys).getBatchIds();
+      doReturn(new URI(path)).when(_context).getRequestURI();
+      doReturn("POST").when(_context).getRequestMethod(); // http method
+      doReturn("CREATE").when(_context).getRestLiRequestMethod(); // header value
```

Comment:
what does this //header value mean?

Reply:
It means that the string returned would come from the "X-RestLi-Method" request header.

Reply:
Updated the inline comment to make it clear.